libs/atqr/domain/src/lib/valueObjects/notification.ts

```diff
@@ -0,0 +1,114 @@
+import { ValueObject } from '../../utils/valueObject';
+import { Guid } from '@tokilabs/lang';
+import { NotificationLogEntry } from './notificationLogEntry.valueObject';
+import { MessageTemplateId } from '../services';
+
+enum NotificationCategory {
+  EuDuvidoProductUpdates = 'EuDuvidoProductUpdates',
+  EuDuvidoPromotions = 'EuDuvidoPromotions',
+  NewInvitations = 'NewInvitations',
+  ChallengeUpdates = 'ChallengeUpdates',
+}
+
+export enum NotificationChannel {
+  Email = 'Email',
+  SMS = 'SMS',
+  Push = 'Push',
+}
+
+export class Notification extends ValueObject<Notification> {
+  constructor(
+    public readonly id: Guid,
+    public readonly category: NotificationCategory,
+    public readonly templateId: MessageTemplateId,
+    public readonly allowedChannels: NotificationChannel[],
+    public readonly createdAt: Date,
+    public readonly requiresUserAction: boolean,
+    public readonly actionTaken: boolean,
+    public readonly contactLog: NotificationLogEntry[],
+    public readonly nextContactDate: Date,
+    public readonly waitingForUserAction: boolean,
+    public messagesSent: number
+  ) {
+    super(Notification, [
+      'category',
+      'templateId',
+      'allowedChannels',
+      'createdAt',
+      'requiresUserAction',
+      'actionTaken',
+      'contactLog',
+      'nextContactDate',
+      'waitingForUserAction',
+      'messagesSent',
+    ]);
+
+    this.id = new Guid();
+  }
+
+  IsWaitingForUserAction(waitingForUserAction: boolean): boolean {
+    if (this.actionTaken === false && this.nextContactDate < new Date()) {
+      return waitingForUserAction == true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
`waitingForUserAction` can never be set because its value is calculated based on other fields. (plus it should have been named `is...` not `Is...`)